### DISCUSSION TITLE
Does pyodide run on any ios based browser?

### DISCUSSION THREAD

**Discussion by ifuchs** [pyodide/pyodide#2149] [Q&A]
I have tried to run https://www.itp.uni-hannover.de/en/groups/weimer/teaching/ideal-gas-simulator/ and other pyodide apps on ios (Iphone 13 and ipad pro) in Safari and Chrome with no success. Is this a current limitation and if so, is it likely that pyodide will run on ios in the near term?

**Comment by hoodmane** [comment]
Well every ios browser uses the same version of Safari (webkit) as its backend. This is required by apple policy. Which version of Safari depends on which version of ios the device runs.

Currently Pyodide runs its test suite in a recent version of Chrome and a recent version of Firefox. We don't run the test suite in any version of Safari, so it's hard for us to detect issues in ios. But even if we added support, we would probably only try to target a recent-ish version of Safari, which would only run on recent-ish apple devices.

**Reply by ifuchs** [reply]
Well, if it ran on "recentish" Apple mobile devices (iPhone, iPad) that would be a lot better than not running there at all. If I deploy an app, there is a high probability that someone will try to run it on one of these devices.

**Comment by ryanking13** [comment]
As hoodmane said, we are not running tests on the Webkit engine, so we can't guarantee whether all functionalities of Pyodide will work in Safari or Chrome in iOS. 

One thing I can tell is that the link you provided uses a quite old version of Pyodide (0.15.0), which was released roughly 2 years ago.
Maybe you can test with a more recent version of Pyodide. The easiest way of testing it is through using [REPL](https://pyodide.org/en/stable/console.html), or [Jupyterlite](https://jupyterlite.readthedocs.io/en/latest/try/lab/).

**Reply by rth** [reply]
Yes, there were definitely Safari related fixes since 0.15.0. As of 0.18.0 the minimal supported version for Safari was 13.1 (cf [docs](https://pyodide.org/en/stable/usage/index.html#supported-browsers)). Though since we are not yet testing minimal versions in CI it's not impossible that there were regressions. 

Generally, we should try to test those in CI (at least for the core tests + a few main packages), and also add Safarii CI https://github.com/pyodide/pyodide/issues/989 which is currently blocked by running CI on Mac  https://github.com/pyodide/pyodide/pull/1912  (though the issue there is more conda related).

**Reply by ifuchs** [reply]
I may have made an error but I tried testing the idealgas.py above with the latest pyodide files (0.19.0) and it did not work on ios. Perhaps someone could verify that this does indeed not work?

**Comment by henryiii** [comment]
FYI, the instance on numpy.org works just fine for me on an iPhone.
